File: csharp/ex_073_meetup/Meetup.cs
using System;

public enum Schedule
{
    Teenth,
    First,
    Second,
    Third,
    Fourth,
    Last
}

public class Meetup
{
    private readonly int _month;
    private readonly int _year;

    public Meetup(int month, int year)
    {
        this._month = month;
        this._year = year;
    }

    public DateTime Day(DayOfWeek dayOfWeek, Schedule schedule)
    {
        DateTime firstDayOfMonth = new DateTime(_year, _month, 1);
        int daysInMonth = DateTime.DaysInMonth(_year, _month);

        switch (schedule)
        {
            case Schedule.Teenth:
                for (int day = 13; day <= 19; day++)
                {
                    DateTime date = new DateTime(_year, _month, day);
                    if (date.DayOfWeek == dayOfWeek)
                        return date;
                }
                break;

            case Schedule.First:
                return GetNthDayOfWeek(1, dayOfWeek);

            case Schedule.Second:
                return GetNthDayOfWeek(2, dayOfWeek);

            case Schedule.Third:
                return GetNthDayOfWeek(3, dayOfWeek);

            case Schedule.Fourth:
                return GetNthDayOfWeek(4, dayOfWeek);

            case Schedule.Last:
                for (int day = daysInMonth; day > 0; day--)
                {
                    DateTime date = new DateTime(_year, _month, day);
                    if (date.DayOfWeek == dayOfWeek)
                        return date;
                }
                break;

            default:
                throw new ArgumentOutOfRangeException(nameof(schedule), "Invalid schedule");
        }

        throw new InvalidOperationException("Date not found");
    }

    private DateTime GetNthDayOfWeek(int n, DayOfWeek dayOfWeek)
    {
        DateTime firstDayOfMonth = new DateTime(_year, _month, 1);
        int dayOffset = ((int)dayOfWeek - (int)firstDayOfMonth.DayOfWeek + 7) % 7;
        int targetDay = 1 + dayOffset + (n - 1) * 7;

        if (targetDay > DateTime.DaysInMonth(_year, _month))
            throw new ArgumentOutOfRangeException(nameof(n), "Invalid week number");

        return new DateTime(_year, _month, targetDay);
    }
}

File: javascript/ex_050_meetup/meetup.js
const Schedule = Object.freeze({
  Teenth: "teenth",
  First: "first",
  Second: "second",
  Third: "third",
  Fourth: "fourth",
  Last: "last"
});

export const meetup = (year, month, schedule, dayOfWeek) => {
  // Convertir día de la semana a índice (0 = Domingo, 1 = Lunes, ..., 6 = Sábado)
  const targetDay = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"].indexOf(dayOfWeek);
  // Crear un array con las fechas del mes que coinciden con el día de la semana
  const dates = [];
  const date = new Date(year, month - 1, 1); // Mes en JavaScript es 0-indexado
  while (date.getMonth() === month - 1) {
    if (date.getDay() === targetDay)
      dates.push(new Date(date)); // Clonar la fecha
    date.setDate(date.getDate() + 1); // Avanzar al día siguiente
  }
  // Determinar la fecha según el schedule
  switch (schedule) {
    case Schedule.First:
      return dates[0];
    case Schedule.Second:
      return dates[1];
    case Schedule.Third:
      return dates[2];
    case Schedule.Fourth:
      return dates[3];
    case Schedule.Last:
      return dates[dates.length - 1];
    case Schedule.Teenth:
      return dates.find(d => d.getDate() >= 13 && d.getDate() <= 19);
    default:
      throw new Error("Invalid schedule");
  }
};